.github/workflows/tests-pr.yml

```diff
@@ -0,0 +1,54 @@
+name: BNB tests on Pull Requests
+
+on:
+  workflow_dispatch:
+    inputs:
+      pr_number:
+        description: 'PR number'
+        required: true 
+        
+env:
+  # To be able to run tests on CUDA 12.2
+  NVIDIA_DISABLE_REQUIRE: "1"
+  SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+  OUTPUT_SLACK_CHANNEL_ID: "C071049S8EM"
+
+jobs:
+  run_tests_single_gpu:
+    strategy:
+      fail-fast: false
+    runs-on: [self-hosted, single-gpu, nvidia-gpu, t4, ci]
+    container:
+      image: "huggingface/peft-gpu-bnb-latest:latest"
+      options: --gpus all --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Pip install and checkout
+        run: |
+          source activate peft
+          pip uninstall bitsandbytes
+          git remote add upstream https://github.com/TimDettmers/bitsandbytes.git
+          git fetch upstream pull/${{ github.event.inputs.pr_number }}/head:test-pr
+          git checkout test-pr
+          cmake -B . -DCOMPUTE_BACKEND=cuda -S . && cmake --build . && pip install -e . && pip freeze | grep bitsandbytes
+          pip install pytest-reportlog pytest-cov parameterized datasets scipy einops
+          pip install "pytest>=7.2.0,<8.0.0" # see: https://github.com/huggingface/transformers/blob/ce4fff0be7f6464d713f7ac3e0bbaafbc6959ae5/setup.py#L148C6-L148C26
+      
```

Comment:
For sake of debugging, highly recommend adding a `pip freeze` step here

Reply:
Makes sense ! done !